src/presentation/ui/page/databases.templ

```diff
@@ -0,0 +1,299 @@
+			createDatabaseUser(databaseType) {
+				htmx.ajax(
+					'POST',
+					'/api/v1/database/' + databaseType + '/' + this.database.name + '/user/',
+					{
+						swap: 'none',
+						values: {
+							username: this.databaseUser.username,
+							password: this.databaseUser.password,
+							privileges: this.databaseUser.privileges,
+						}
+					},
+				);
+				this.closeCreateDatabaseUserModal();
+			},
+			isDeleteDatabaseUserModalOpen: false,
+			openDeleteDatabaseUserModal(databaseName, databaseUsername) {
+				this.resetPrimaryStates();
+
+				this.database.name = databaseName;
+				this.databaseUser.username = databaseUsername;
+				this.isDeleteDatabaseUserModalOpen = true;
+			},
+			closeDeleteDatabaseUserModal() {
+				this.isDeleteDatabaseUserModalOpen = false;
+			},
+			deleteDatabaseUserElement(databaseType) {
+				htmx.ajax(
+					'DELETE',
+					'/api/v1/database/' + databaseType + '/' + this.database.name + '/user/' + this.databaseUser.username + '/',
+					{swap: 'none'},
+				);
+				this.closeDeleteDatabaseUserModal();
+			},
+		}))
+	})
+}
+
+templ DatabasesIndex(selectedDbTypeSummary presenterDto.SelectedDatabaseTypeSummary) {
+	@DatabasesIndexLocalState()
+	<div class="flex flex-col" x-data="databases">
+		<div class="mb-6 flex flex-row items-center justify-between">
+			@componentStructural.PageTitle(
+				"Databases", "Manage your database services by adding and deleting databases or users.", "ph-database",
+			)
+		</div>
+		<div
+			id="databases-content"
+			hx-get={ "/databases/?dbType=" + selectedDbTypeSummary.Type.String() }
+			hx-trigger="submit from:form delay:500ms, click from:button#delete-element delay:500ms, click from:button#create-database-user delay:500ms"
+			hx-select="#databases-content"
+			hx-target="#databases-content"
+			hx-swap="outerHTML transition:true"
+			class="card w-full"
+		>
+			<div id="databases-tabs">
+				@DatabasesTabs(selectedDbTypeSummary)
+			</div>
+			@CreateDatabaseModal(selectedDbTypeSummary.Type)
+			@componentStructural.DeleteModal(
+				"isDeleteDatabaseModalOpen", "closeDeleteDatabaseModal()",
+				"deleteDatabaseElement('"+selectedDbTypeSummary.Type.String()+"')",
+				"database.name", "",
+			)
+			@CreateDatabaseUserModal(selectedDbTypeSummary.Type, selectedDbTypeSummary.Databases)
+			@componentStructural.DeleteModal(
+				"isDeleteDatabaseUserModalOpen", "closeDeleteDatabaseUserModal()",
+				"deleteDatabaseUserElement('"+selectedDbTypeSummary.Type.String()+"')",
+				"databaseUser.username", "database.name",
+			)
+		</div>
+	</div>
+}
+
+func getDatabasesTabHeaderItems(
+	selectedDatabaseType valueObject.DatabaseType,
+) []componentStructural.TabHeaderItem {
+	tabHeaderItems := []componentStructural.TabHeaderItem{
+		{Label: "MySQL/MariaDB/Percona", ComponentValue: "mariadb"},
+		{Label: "PostgreSQL", ComponentValue: "postgresql"},
+	}
+
+	for itemIndex := range tabHeaderItems {
+		if selectedDatabaseType.String() != tabHeaderItems[itemIndex].ComponentValue {
```

Comment:
This is a pure Go code, why are you casting .String() all the time instead of having a selectedDbTypeStr before the for?

Reply:
I moved the casting outside of the `for` loop.